Fix part of #4057: Added a test file for exploration-language-code.service.ts, exploration-title.service.ts, exploration-objective.service.ts, exploration-init-state-name.service.ts

safwan20 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This fixes part of #4057.
2. This PR does the following: Added test files for exploration-language-code.service.ts, exploration-title.service.ts, exploration-objective.service.ts, exploration-init-state-name.service.ts

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

safwan20 committed:
test-1

safwan20 committed:
Merge branch 'develop' of github.com:oppia/oppia into exploration-tests

safwan20 committed:
test-2

safwan20 committed:
test-3

safwan20 committed:
test-4

kevintab95 commented:
Hi @safwan20, just a general note. I see that you have 5 open PRs ATM. Our recommendation to contributors is to have only 1-2 active PRs at any point. They can still work on other issues in a local branch but create a PR only once your other PRs are merged. This is an effort to reduce the load on the CI queue.

Btw, you can combine your PRs as a workaround, if your changes are related.

Tagging @kevinlee12 to add anything that I might have missed. Thanks!

kevintab95 commented on scripts/check_frontend_coverage.py:
```diff
@@ -129,6 +129,8 @@
     'exploration-id-validation.service.ts',
     'exploration-init-state-name.service.ts',
+    'exploration-language-code.service.ts',
+    'exploration-objective.service.ts',
```
What about the other two services? Are they not fully covered in this PR?

safwan20 replied:
![Screenshot from 2020-04-24 01-41-05](https://user-images.githubusercontent.com/45730556/80144883-eaf57980-85cc-11ea-975c-35b0cd8d2117.png)

all are covered.

kevintab95 reviewed (commented):
Left a comment. PTAL @safwan20. Thanks!

kevinlee12 commented:
Nothing to add, thanks for pointing this out! I really want to stress the part about increasing pressure on our CI. Our wait times are now 6+ hours, so we really need everyone’s help to keep it down. 

> On Apr 21, 2020, at 6:30 PM, Kevin Thomas <notifications@github.com> wrote:
> 
> ﻿
> Hi @safwan20, just a general note. I see that you have 5 open PRs ATM. Our recommendation to contributors is to have only 1-2 active PRs at any point. They can still work on other issues in a local branch but create a PR only once your other PRs are merged. This is an effort to reduce the load on the CI queue.
> 
> Btw, you can combine your PRs as a workaround, if your changes are related.
> 
> Tagging @kevinlee12 to add anything that I might have missed. Thanks!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

U8NWXD committed:
Fix #9086 and #9100: Fix classroom e2e flakes (#9118)

* Wait for Topic Thumbnail Visibility Before Editing

Before editing the topic thumbnail, wait for it to be visible.

Fixes #9086.

* Remove Duplicate Create Topic Button HTML Tag

When no topics have been created, the topics and skills dashboard page
shows a create topic button in addition to the button in the navigation
bar. These two buttons shared the `protractor-test-create-topic-button`
HTML tag, which caused Protractor to raise warnings. This commit removes
the HTML tag from the button outside the navigation bar.

* First Wait for Role Update Confirmation Visibility

We were waiting for the role update confirmation message to first
contain the expected text and then to be visible. This commit swaps the
order of those checks in an attempt to address #9100 and #9097.

safwan20 committed:
Fix part of #4057: Added a test file for  exploration-param-changes.service.ts (#9114)

* test-1

* test-2

* test-3

* test-4

abhayraghuwanshi committed:
Fix part of #4057. Unit test for exploration-feature-service.ts (#9076)

* made service file

* the exploration data object changed

* service test added

* added to 100% coverage

* sorted alphabtically

* changed the exploration feature tests

* requested change done

* fixed the linter tests

kevinlee12 commented:
@safwan20 mind the frontend test failure!

bansalnitish committed:
Fix part of #8472: Upgrades CollectionCreationBackendApiService to Angular (#9125)

* Upgrades CollectionCreationBackendApiService to Angular

* addresses comments

DubeySandeep committed:
Add a lint check for all jobs register in jobs_registry.py file (#9129)

* Add test for job registry

* Fix removal issue.

* Removed unwanted try-catch

* Address review comments.

* Removed test for non-existing one-off job

* Removed unwanted newline!

* Address review comments (2)

* Undo unwanted changes

* Removed unwanted tests.

safwan20 commented:
> Hi @safwan20, just a general note. I see that you have 5 open PRs ATM. Our recommendation to contributors is to have only 1-2 active PRs at any point. They can still work on other issues in a local branch but create a PR only once your other PRs are merged. This is an effort to reduce the load on the CI queue.
> 
> Btw, you can combine your PRs as a workaround, if your changes are related.
> 
> Tagging @kevinlee12 to add anything that I might have missed. Thanks!

I will keep this in mind.

aks681 committed:
Fix #9121: Address review comments in #9087 (#9126)

* fix #9121

* made review  changes

* review change

aks681 committed:
Fix UI and UX issues in new structures editors and dashboard (#9110)

* Fix issues in new structures editors

* e2e test

* minor changes

* fix cramped skill cards

* fixed e2e test: added tab management

* rename value

* made review changes

* made review change

tjinjoy committed:
Fixes issue #3369: Resize editor to avoid overflowing when creator uses longer responses (#9122)

* Resize contenteditable textbox to prevent response overflow #3369

* Changed spacing in ckeditor directive

* Fix errors from lint and typescript tests

* Fix quoted width property lint error

brianrodri committed:
Introduce contextmanager for creating temporary directories (#9142)

* introduce linter_utils.temp_dir

* fix bad-continuation

* fix lint issues

* fix typos

* move to end of function

ankita240796 committed:
Add check for invalid emails (#9130)

* Add check for invalid emails

* Fix naming and messages

* Fix tests

* Update email check for only new contributors and authors

* Fix docstrings

* Fix emails

* Add error message, fix name

* Fix name

Commit:
Fix part of #8015 : Domain object for topic-viewer-backend-api.service.ts (#8975)

* Returning domain object from backend

* getting all the changes for domain-object

* getting all the changes for domain-object

addressing requested changes

* Making changes for tests

* Added tests for additional_story_dicts

* Added tests for additional_story_dicts

Sorted order for UpgradedService.ts

* minor changes

brianrodri committed:
dedupe ladbas (#9151)

safwan20 committed:
Fix part of #4057: Added a test file for exploration-category-service.ts (#9108)

* test-1

* added to coverage

* test-2

* test-3

* test-4

* test-5

* test-6

* test-7

* add change

* test-5

* added let

* added let-1

varun-tandon committed:
Fix part of #5914: Complete Controller Layer of Takeout (#9133)

* first pass controller layer

* linting

* linting

* linting

* update to literal

* linting

* switch to epoch

* switch to epoch

* resolve linting

* resolve linting

* resolve linting

* added tests for more cases

* linting

* remove mktime

* linting

* switch to constant

marianadasilvadev committed:
Remove JQuery css manipulation from controller and use ng-class in template instead (#9104)

* Remove JQuery css class manipulation from controller and use ng-class in template instead

* Fix review comments

* Fix review comments

* Fix review comments

* Fix review comments

* Fix review comments

* Fixing onhashchange bug

* Undo changes

ankita240796 committed:
Add check for version length (#9147)

* Add check for version length

* Fix error message

marianadasilvadev reviewed (approved):
LGTM thanks @safwan20

marianadasilvadev commented:
Actually, it seems exploration title services are failing the frontend tests, please take a look and fix it.

nithusha21 committed:
Fix #9079: Merge the target branch before running the tests on CircleCI (#9084)

* try to get the target branch

* try different env variables

* do the git merge

* include command after all checkouts

* update version

* add git user

* add no-edit flag to suppress editor

* trying to debug why target_branch is null

* Remove debugging logs

* attempt to debug lint tests

* remove all other suites

* add all suites back to prevent build errors

* Remove jq after parsing target branch

* revert changes in pre_commit_linter

nishantwrp committed:
Add a second function argument to avoid the warning message (#9160)

Arnesh07 committed:
Fixes #7448: Adds a filter option to filter skills. (#9082)

* adds filter for selecting skills

* adds unit test to test the filter in reader.py

* docstring changes

* uses skill_id instead of i in for loop

U8NWXD committed:
Transfer CODEOWNER of /core/tests to @nithusha21 (#9166)

@U8NWXD is familiar with the e2e tests, but not the rest of /core/tests.

safwan20 committed:
Merge branch 'exploration-tests' of github.com:safwan20/oppia into exploration-tests

safwan20 committed:
added-title
